### ISSUE TITLE
Should flatListToHierarchical default params correspond to current content block props

### ISSUE THREAD

**Issue by JEverhart383** [wpengine/faustjs#1506]
When implementing a new project using Faust and WPGraphQL content blocks, I stumbled when converting my list of blocks into a tree because the [default client and parent id keys for the function](https://github.com/wpengine/faustjs/blob/1573bf95f92da53c5774d69f35584e731ea88bb8/packages/faustwp-core/src/utils/flatListToHierarchical.ts#L16C24-L16C24) are _different_ from what content blocks return. 

We should consider changing the default params to correspond to what GraphQL will return so that no additional work is needed beyond passing in the array of blocks.

In addition, [this doc](https://github.com/wpengine/faustjs/blob/1573bf95f92da53c5774d69f35584e731ea88bb8/packages/faustwp-core/src/utils/flatListToHierarchical.ts#L16C24-L16C24) should be updated to reflect the updated properties that WPGraphQL returns related to ids and show the need to pass an additional param to this utility function:
```
const blocks = flatListToHierarchical(editorBlocks, {idKey: 'clientId', parentKey: 'clientParentId'});
```

It looks like the docs also include an example where `clientId` is aliased as `id`, which is maybe another way to approach this in examples, but I'm not sure we can assume that everyone coming to Faust will know what that means and how to do it in GraphQL parlance. If we do go that route, we should call out why we're doing this and link to [the GraphQL docs on aliasing](https://graphql.org/learn/queries/#aliases).

**Comment by theodesp** [comment]
Hey @JEverhart383 Changing the defaults could mean a breaking change since it would break existing clients using the `flatListToHierarchical`. Maybe we should document the `flatListToHierarchical` better instead of bumping a major version.

**Comment by theodesp** [comment]
also `flatListToHierarchical` is used in many places and not only in handling blocks. For example when rendering menus and such.

**Comment by JEverhart383** [comment]
You are absolutely right @theodesp about the breaking change. I thought about that right after I posted this comment. And I wasn't aware that it had other uses, so that is another reason not to change the API. Forget my suggestion there. I think either showing how to call the function with override ids or recommending aliases in the query are both good options, and maybe showing people both and letting them decide.

**Comment by josephfusco** [comment]
Closing — as discussed in the thread, the original author agreed the default params shouldn't change since it would be a breaking change (and `flatListToHierarchical` is used beyond just blocks, e.g. for menus).

The documentation around content blocks and `flatListToHierarchical` usage has been improved since this was filed. Thanks @JEverhart383 for the thoughtful suggestion and discussion!
